internal/handler/msg_saslstart.go

```diff
@@ -78,6 +82,13 @@ func (h *Handler) MsgSASLStart(ctx context.Context, msg *wire.OpMsg) (*wire.OpMs
 		)))
 
 	case "SCRAM-SHA-1", "SCRAM-SHA-256":
+		if !h.EnableNewAuth {
```

Comment:
@henvic see this change, highlighting SCRAM is only enabled with new authentication